### ISSUE TITLE
[ISSRE'2014] Assessing the Threat Landscape for Software Libraries 

### ISSUE THREAD

**Issue by vivekaxl** [opensciences/opensciences.github.io#161]
Although a majority of the code in numerous applications comes from libraries, the risk of security vulnerabilities that comes with these libraries is often overlooked. In this regard, we seek to assess the threat landscape associated with software libraries and discuss mitigation strategies via Security Development Lifecycle (SDL).

Common Attack Pattern Enumeration and Classification (CAPEC) is a publicly available, community- developed list of common attack patterns along with a comprehensive schema and classification taxonomy [8]. Each attack pattern captures knowledge about how specific parts of an attack are designed and executed, providing the attacker’s perspective on the problem and the solution, and gives guidance on ways to mitigate the attack’s effectiveness.

Bottom line: The author looks at 453 documented patterns  and enumerated the associated severity, likelihood of exploit and attacker skill required as documented by CAPEC, eventually identifying 27 attack patterns.

Data link: http://capec.mitre.org/documents/schema/schema_v2.7.html

Emails:
luca.recchia@mbda.it
giuseppe.procopio@mbda.it

Since this is a publicly available data...I don't think you need to write an email.

**Comment by CarterPape** [comment]
Paper link: http://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=6983805

**Comment by BenProvince** [comment]
There are 19 versions (to date) of this database, along with schema and definitions available from the MITRE web site. I'm not so sure that this need to be duplicated on PROMISE, but perhaps we should link to it?

Data, content, schemas, and documentation for all versions:
http://capec.mitre.org/data/index.html#downloads

Latest schema documentation:
http://capec.mitre.org/documents/schema/schema_v2.7.html

**Comment by rahlk** [comment]
### Categorization (effort, requirements, model, defect, etc.)
    If it doesn't fit nicely into one of these categories, put "other" and propose a category name
    Pick from the categories in the navbar on the OpenScience website.

### Author list for the paper (including emails)

### General overview of the data
    Note that this is not the abstract of the paper. If you can find an overview of the dataset in the author's own words, just copy those words. If not, make your own overview. Paragraph long (if you are making it yourself)

### Link to the paper associated with the dataset
    This should be either a link to the ACM digital library (e.g. http://dl.acm.org/citation.cfm?id=2635868.2635905), IEEE digital library (e.g. http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=6982619), or some other standard library

### Paper abstract

#### BibTeX reference for the paper

### Link to the datasets
    If multiple files come with it, include links to all files, not just the parent file
    Red flags should go up if, after downloading all the data, it is larger than 100 MB (compressed)

### Is this dataset part of a larger series or collection?
    If so, link to the master index of the series or collection (if possible)

**Comment by reesjones** [comment]
Added as [capec](http://openscience.us/repo/other/capec.html), data being uploaded.

**Comment by reesjones** [comment]
Author emailed.
